lib/Twig/Environment.php

```diff
@@ -551,13 +551,20 @@ public function setCompiler(Twig_Compiler $compiler)
     /**
      * Compiles a node and returns the PHP code.
      *
-     * @param Twig_Node $node A Twig_Node instance
+     * @param Twig_Node   $node   A Twig_Node instance
+     * @param string|null $source The template source code
      *
      * @return string The compiled PHP source code
      */
-    public function compile(Twig_Node $node)
+    public function compile(Twig_Node $node, $source = null)
```

Comment:
should the signature of this method really be changed, or should it be added only in ``compileSource`` ?